localstack/aws/proxy.py

```diff
@@ -28,7 +32,11 @@ def get_account_id_from_request(request: Request) -> str:
         extract_access_key_id_from_auth_header(request.headers) or TEST_AWS_ACCESS_KEY_ID
     )
     set_aws_access_key_id(access_key_id)
-    return get_account_id_from_access_key_id(access_key_id)
+
+    account_id = get_account_id_from_access_key_id(access_key_id)
+    set_aws_account_id(account_id)
```

Comment:
IMO the `set_*` calls should be extracted. Would make this easier to test in isolation and more predictable in terms of behavior.

Reply:
Not sure I understand -- could you clarify what you mean by extracting the `set_*` calls?

Reply:
Fair point. The name `get_account_id_from_request` gives the wrong impression. Will fix it separately, thanks :+1: